nevergrad/optimization/optimizerlib.py

```diff
@@ -1668,3 +1715,36 @@ def __init__(self, parametrization: IntOrParameter, budget: Optional[int] = None
                     self.optims = [CMA(self.parametrization, budget, num_workers)]
             else:
                 self.optims = [NGO(self.parametrization, budget, num_workers)]
+
+
+@registry.register
+class MetaModel(base.Optimizer):
+    """Adding a metamodel into CMA."""
+
+    def __init__(self, parametrization: IntOrParameter, budget: Optional[int] = None, num_workers: int = 1) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        assert budget is not None
+        self.optims = [
+            Shiwa(self.parametrization, budget, num_workers),  # share parametrization and its rng
+        ]  # noqa: F405
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        # We request a bit more points than what is really necessary for our dimensionality (+dimension).
+        if (self._num_ask % max(self.num_workers, self.dimension) == 0 and
+                len(self.archive) >= (self.dimension*(self.dimension-1))/2 + 2*self.dimension + 1):
+            try:
+                data = learn_on_k_best(self.archive, int((self.dimension*(self.dimension-1))/2 + 2*self.dimension + 1))
+                candidate = self.parametrization.spawn_child().set_standardized_data(data)
+            except InfiniteMetaModelOptimum:  # The optimum is at infinity. Shit happens.
+                candidate = self.optims[0].ask()
+        else:
+            candidate = self.optims[0].ask()
+        return candidate
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, value: float) -> None:
+        self.optims[0].tell(candidate, value)
+
+    #def _internal_tell_not_asked(self, candidate: p.Parameter, value: float) -> None:
```

Comment:
I have no idea what I'm doing here.

Reply:
MetaModel works with CMA as a base algorithm, but not with Shiwa -- in the latter case it will raise an error because of TellNotAsked.

Reply:
all Shiwa cases do not handle TellNotAsked (eg Powell) so it seems normal that it does not work